src/libs/ajax/fetch/fetch-core.ts

```diff
@@ -49,3 +50,101 @@ export const fetchOk = _.flow(withInstrumentation, withCancellation, withErrorRe
 export const withUrlPrefix = _.curry((prefix, wrappedFetch) => (path, ...args) => {
   return wrappedFetch(prefix + path, ...args);
 });
+
+export interface MakeWithRetryOnErrorArgs {
+  /** i.e. 5000 - the maximum elapsed timeout (in milliseconds) for sum of all request attempts */
+  maxTimeout: number;
+
+  /** i.e. 1500 - the maximum attempt random delay (in milliseconds) that can occur before a request retry */
+  maxAttemptDelay: number;
+
+  /** i.e. 500 - the minimum attempt random delay (in milliseconds) that can occur before a request retry */
+  minAttemptDelay: number;
+}
+
+/**
+ * factory method for withRetryOnError.  Most consumers will likely use pre-made withRetryOnError function
+ * @param makeArgs
+ */
+export const makeWithRetryOnError = (makeArgs: MakeWithRetryOnErrorArgs) => {
+  const withRetryOnErrorWrapper =
+    (shouldNotRetryFn: (error: unknown) => boolean) =>
+    (wrappedFetch: FetchFn) =>
+    async (...args: Parameters<FetchFn>) => {
+      const timeout = makeArgs.maxTimeout;
+      const somePointInTheFuture = Date.now() + timeout;
+      const maxDelayIncrement = makeArgs.maxAttemptDelay;
+      const minDelay = makeArgs.minAttemptDelay;
+
+      while (Date.now() < somePointInTheFuture) {
+        try {
+          await delay(minDelay + maxDelayIncrement * Math.random());
+          return await wrappedFetch(...args);
+        } catch (error) {
+          if (shouldNotRetryFn(error)) {
+            throw error;
+          }
+          // else: ignore error, will retry
+        }
+      }
+      return wrappedFetch(...args);
+    };
+  return withRetryOnErrorWrapper;
+};
+
+/**
+ * creates a fetch function augmenter with logic to keep retrying a request that throws an error.  A request will be
```

Comment:
Creates a fetch function augmenter that retries a failed request based on the return value of `shouldNotRetryFn`.

Reply:
I like these suggestions, will get them in shortly.